src/containerapp/azext_containerapp/containerapp_sessionpool_decorator.py

```diff
@@ -0,0 +1,379 @@
+        else:
+            if environment_name is not None:
+                raise ValidationError(f"Do not pass environment name when using container type {container_type}")
+
+    def construct_payload(self):
+        self.session_pool_def["location"] = self.get_argument_location()
+
+        # We only support 'Dynamic' type in CLI
+        self.session_pool_def["properties"]["poolManagementType"] = "Dynamic"
+        self.session_pool_def["properties"]["environmentId"] = self.get_argument_managed_env()
+        if self.get_argument_container_type() is None:
+            self.set_argument_container_type(ContainerType.PythonLTS.name)
+        self.session_pool_def["properties"]["containerType"] = self.get_argument_container_type()
+
+        dynamic_pool_def = self.set_up_dynamic_configuration()
+        session_network_def = self.set_up_network_configuration()
+        session_scale_def = self.set_up_scale_configuration()
+        secrets_def = self.set_up_secrets()
+
+        # CustomerContainerTemplate
+        customer_container_template = None
+        if self.get_argument_container_type() == ContainerType.CustomContainer.name:
+            customer_container_template = {}
+            self.validate_environment()
+
+            container_def = self.set_up_container()
+            ingress_def = self.set_up_ingress()
+            registry_def = self.set_up_registry_auth_configuration(secrets_def)
+
+            customer_container_template["containers"] = [container_def]
+            customer_container_template["ingress"] = ingress_def
+            customer_container_template["registryCredentials"] = registry_def
+
+        safe_set(self.session_pool_def, "properties", "customContainerTemplate", value=customer_container_template)
+        safe_set(self.session_pool_def, "properties", "secrets", value=secrets_def)
+        safe_set(self.session_pool_def, "properties", "dynamicPoolConfiguration", value=dynamic_pool_def)
+        safe_set(self.session_pool_def, "properties", "sessionNetworkConfiguration", value=session_network_def)
+        safe_set(self.session_pool_def, "properties", "scaleConfiguration", value=session_scale_def)
+
+    def set_up_dynamic_configuration(self):
+        dynamic_pool_def = {}
+        dynamic_pool_def["executionType"] = "Timed"
+        if self.get_argument_cooldown_period_in_seconds() is None:
+            self.set_argument_cooldown_period_in_seconds(300)
+        dynamic_pool_def["cooldownPeriodInSeconds"] = self.get_argument_cooldown_period_in_seconds()
+        return dynamic_pool_def
+
+    def set_up_network_configuration(self):
+        session_network_def = {}
+        if self.get_argument_egress_enabled() is None:
+            self.set_argument_egress_enabled(False)
+        session_network_def['status'] = convert_egress_parameter(self.get_argument_egress_enabled())
+        return session_network_def
```

Comment:
I think finish set `session_network_def` in `set_up_network_configuration` would be better:
`safe_set(self.session_pool_def, "properties", "sessionNetworkConfiguration", value=session_network_def)`

Reply:
Like discussed offline, I would like to keep it. thank you.

Reply:
Would like to ask your favor to resolve.